lua/entities/gmod_wire_expression2/core/npc.lua

```diff
@@ -7,70 +7,84 @@ E2Lib.RegisterExtension("npc", true, "Allows controlling of NPCs.", "NPCs can be
 __e2setcost(5) -- temporary
 
 local function validNPC(entity)
-	return IsValid(entity) && entity:IsNPC()
+	return IsValid(entity) and entity:IsNPC()
 end
 
 e2function void entity:npcGoWalk(vector rv2)
-	if !validNPC(this) || !isOwner(self,this) then return end
+	if not validNPC(this) then return self:throw("Entity e: is not a valid NPC!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this NPC!", nil) end
+
 	this:SetLastPosition( Vector(rv2[1], rv2[2], rv2[3]) )
 	this:SetSchedule( SCHED_FORCED_GO )
 end
 
 e2function void entity:npcGoRun(vector rv2)
-	if !validNPC(this) || !isOwner(self,this) then return end
+	if not validNPC(this) then return self:throw("Entity e: is not a valid NPC!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this NPC!", nil) end
+
 	this:SetLastPosition( Vector(rv2[1], rv2[2], rv2[3]) )
 	this:SetSchedule( SCHED_FORCED_GO_RUN )
 end
 
 e2function void entity:npcAttack()
-	if !validNPC(this) || !isOwner(self,this) then return end
+	if not validNPC(this) then return self:throw("Entity e: is not a valid NPC!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this NPC!", nil) end
+
 	this:SetSchedule( SCHED_MELEE_ATTACK1 )
 end
 
 e2function void entity:npcShoot()
-	if !validNPC(this) || !isOwner(self,this) then return end
+	if not validNPC(this) then return self:throw("Entity e: is not a valid NPC!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this NPC!", nil) end
+
 --	if !this:HasCondition( 6 ) then return end -- COND_NO_WEAPON. See http://maurits.tv/data/garrysmod/wiki/wiki.garrysmod.com/index4389.html
 	this:SetSchedule( SCHED_RANGE_ATTACK1 )
 end
 
 e2function void entity:npcFace(vector rv2)
-	if !validNPC(this) || !isOwner(self,this) then return end
+	if not validNPC(this) then return self:throw("Entity e: is not a valid NPC!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this NPC!", nil) end
+
 	local Vec = Vector(rv2[1], rv2[2], rv2[3]) - self.entity:GetPos()
 	local ang = Vec:Angle()
-	this:SetAngles( Angle(0,ang.y,0) )
+	this:SetAngles( Angle(0, ang.y ,0) )
 end
 
 e2function void entity:npcGiveWeapon()
-	if !validNPC(this) || !isOwner(self,this) then return end
+	if not validNPC(this) then return self:throw("Entity e: is not a valid NPC!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this NPC!", nil) end
 
 	local weapon = this:GetActiveWeapon()
-	if (weapon:IsValid()) then
-		if (weapon:GetClass() == "weapon_smg1") then return end
+	if weapon:IsValid() then
+		if weapon:GetClass() == "weapon_smg1" then return end
```

Comment:
Shouldn't this be ai_weapon_smg1?

Reply:
Printed out weapon:GetClass() and it was weapon_smg1 (Also made sure to concat a string to it so garry prints wouldn't be deceiving me). Also tested using weapon_smg1 and got the same result.

I have no idea why we use ai_weapon_smg1 but I'm just gonna assume :Give() just allows ai_ as a prefix. Or maybe ai_ weapons are restricted to the default ones.